src/runtime/halide_hexagon_dma_api.cpp

```diff
@@ -0,0 +1,266 @@
+    //////////////////////////////////////////////////////////////////////////
+    handle = halide_hexagon_dmart_get_dma_handle(user_context, dma_handle, frame);
+    if(handle == 0) {
+        error(user_context) << "Function failed to get DMA Write  Handle  \n";
+        return HEX_ERROR;
+    }
+
+    if(handle != 0) {
+        dma_finish_frame(handle);
+        uintptr_t fold_addr = halide_hexagon_dmart_get_fold_addr(user_context, dma_handle, frame);
+        halide_hexagon_dma_wrap_memory_free(user_context, dma_handle, (void*)fold_addr);
+        halide_hexagon_dmart_detach_context(user_context, dma_handle, frame);
+        dma_free_dma_engine(handle);
+    }
+    buf->device_interface->impl->release_module();
+    buf->device = 0;
+    buf->device_interface = NULL;
+
+    return HEX_SUCCESS;
+}
+
+/**
+ * halide_hexagon_dmart_get_memory
+ * desc: Calculate width, height, stride and fold from roi buf
+ * Need also type and padding information. How ???
+ * Allocate Memory
+ */
+void* halide_hexagon_dmaapp_get_memory(void* user_context, halide_buffer_t *roi_buf) {
+    void *vret = halide_hexagon_dma_memory_get(user_context, roi_buf);
+
+    if(vret == NULL) {
+        int nCircularFold;
+        int w,h,s;
+        void* vret = NULL;
+
+        halide_hexagon_dma_user_component_t comp;
+        if(halide_hexagon_dma_comp_get(user_context, roi_buf, comp)) {
+            error(user_context) << "Failure to get the component" ;
+            return NULL;
+        }
+
+        /////////////////////////////////////////////////////
+        //Problem with not including the inframe buf
+        // Have to take the global version of dma context
+        /////////////////////////////////////////////////////
+        HexagonDmaContext hexagon_dma_context(user_context);
+        halide_assert(user_context, hexagon_dma_context.context != NULL);
+        dma_context dma_ctxt = hexagon_dma_context.context;
+
+        // ASSUMPTION: no folding
+        nCircularFold = 1;
+        // Divide Frame to predefined tiles in Horizontal Direction
+        w = roi_buf->dim[0].extent;
+        // Divide Frame to predefined tiles in Vertical Direction
+        h = roi_buf->dim[1].extent;
+        // Each tile is again vertically split in to predefined DMA transfers
+        // Stride is aligned to Predefined Value
+        s = roi_buf->dim[1].stride;
+     
+        /////////////////////////////////////////////////////////
+        //ASSUMPTION: Assuming paddng and fmt type to be default here since we really do  not have the inframe
+        //int halide_hexagon_dmart_get_padding(void* user_context, uintptr_t frame);
+        //int halide_hexagon_dmart_get_fmt_type(void* user_context, uintptr_t frame)
+        ///////////////////////////////////////////////////
+        int padding = 0;
+        int type = NV12;
+
+        // allocate L2$
+        vret = halide_hexagon_dma_memory_alloc(user_context, dma_ctxt, comp, w, h, s, nCircularFold, padding, type);

```

Comment:
Please consider rewriting this as
`vret = halide_hexagon_dma_memory_alloc(user_context, dma_ctxt, comp, roi_buf->dim[0].extent,
                                                                          roi_buf->dim[1].extent, roi_buf->dim[1].stride,
                                                                           nCircularFold, padding, type);`

This makes it easy to see what is being passed.